Created a window of multivariate rules plot.

heberlr opened the pull request:
- Added a button in the rules tab to plot the multivariate rules.

- Added the file multivariate_rules.py, which can plot multivariate rules standalone by simply loading the CSV files. However, since the csv file doesn't have the base value of rules, this doesn't represent the right values of rules.

**This has some issues:**
I created the function associated with the button, but the window does not appear. We need to prepare the data from the rules tab for this plot. This is expecting to receive a CSV file name and a dataframe with the following column names: ['cell', 'signal', 'direction', 'behavior', 'saturation', 'half_max', 'hill_power', 'dead', 'base_behavior'].

drbergman commented on bin/multivariate_rules.py:
```diff
@@ -0,0 +1,355 @@
+        value = self.tickValue( self.slider.value() )
+        self.label.setText(f"{self.description}: {value}")
+
+class MyWidget_behavior(QWidget):
```
Can we remove the "My" from these names? Maybe call it `BehaviorWidget`

drbergman commented on bin/multivariate_rules.py:
```diff
@@ -0,0 +1,355 @@
+        Max_behavior_values = ["%e" % x for x in np.linspace(self.max_up_min,self.max_up_max, num=11, endpoint=True)]
+        Min_behavior_values = ["%e" % x for x in np.linspace(self.max_down_min, self.max_down_max, num=11, endpoint=True)]
+        self.slider_base_behavior = LabeledSlider("Base behavior", Base_behavior_values)
+        self.slider_up_behavior = LabeledSlider("Max up regularion", Max_behavior_values)
```
"regularion"

drbergman commented on bin/multivariate_rules.py:
```diff
@@ -0,0 +1,355 @@
+        self.combobox_behavior.clear()
+        list_behaviors = self.CSV_loader.dataframe.loc[self.CSV_loader.dataframe["cell"] == self.combobox_cell.currentText()]['behavior'].tolist()
+        self.combobox_behavior.addItems(list_behaviors)
+    def update_rules(self):
```
<img width="318" alt="image" src="https://github.com/PhysiCell-Tools/PhysiCell-Studio/assets/22732264/1463d55e-9153-4afa-8247-3d7e744c9120">

behavior loaded in twice (possibly because two rules each targeting the same behavior?)

heberlr replied:
Could you send me the rules that you tested?

drbergman replied:
[cell_rules__.csv](https://github.com/PhysiCell-Tools/PhysiCell-Studio/files/15041863/cell_rules__.csv)

drbergman commented on bin/rules_tab.py:
```diff
@@ -1171,6 +1178,11 @@ def response_combobox_changed_cb(self, idx):
         #     return
+    #-----------------------------------------------------------
+    def plot_rules(self):
+        RulesWindow = Window_plot_rules()
```
`self.RulesWindow` will make it actually display. I don't know why...

heberlr replied:
Yeah! This is one thing that I would like some help.

drbergman reviewed (changes requested):
Seems like a promising addition! Just gotta work out some of the bugs integrating with everything. Long term, do you think you want to keep the `Load CSV` button there? Or only use it to read in the rules in the Rules tab? Maybe the answer is just the obvious one: initially load the Rules tab rules and allow the user to load in another set?

rheiland commented:
I want to understand how this is supposed to work before merging. I cloned your branch and ran it, added a couple of rules to the table (with same behavior, but different signals), then click the new `Plot rules` button, but no plotting window appears. Should it? Also, after clicking the new button, the cells in the table seem to be unselectable. Can show you in person sometime.

heberlr commented:
Thank you for the feedback. This is something we can discuss. My idea was allow the user to load another csv, but ultimately you can close this window and load the csv in the Studio and plot again using the right base value. This load csv alone won't use the proper base values. I think you convenced me, I will remove the load csv thing.

heberlr commented:
Everything is working now. However, we still need to create a dynamic dataframe compatible in rules_tab.py. I have included an example there. I addressed all your points, Daniel. Thank you!

rheiland commented:
I've begun looking at the dynamic dataframe. However, I still don't understand the Down regulation slider. It doesn't seem to do anything, for the sample data you provided.  Or maybe I've done something to make it behave that way.

Also, we may want to disable the mouse scrollwheel from moving a Slider when it's in focus under the cursor. Or wait to see if users complain. Personally, I was a little shocked when I accidentally did it.

heberlr commented:
I did the dynamic dataframe in this new commit. I define all the sliders from behavior changing +/- 100% of the real value, if it is zero doesn't change. Any ideas to do it automatically?

rheiland commented:
Thanks for doing the dynamic dataframe! I've played with it a bit more. Are we comfortable with going ahead and exposing this functionality in the Studio in the next release? If so, I will do it this weekend. An alternative is to make it a beta feature (requiring the user to specify an arg on the command line at startup). Do you think Paul is on board with this feature?
